planner/core/exhaust_physical_plans.go

```diff
@@ -1996,6 +1996,7 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		mppShuffleJoin:   !useBCJ,
 		// Mpp Join has quite heavy cost. Even limit might not suspend it in time, so we dont scale the count.
 	}.Init(p.ctx, p.stats, p.blockOffset, childrenProps...)
+	join.SetSchema(p.schema)
```

Comment:
why add this? show some examples? because it will change all join, pls be careful.

Reply:
this is only for tiflash mpp join, and you can see `func BuildPhysicalJoinSchema(joinType JoinType, join PhysicalPlan) *expression.Schema`, all the join schema except left semi family will be rebuild.

Reply:
The schema of left semi family is also rebuild in [`BuildPhysicalJoinSchema`](https://github.com/pingcap/tidb/blob/a75520e259c4f42f2dd2e4e24d23196ce36967fc/planner/core/util.go#L245), so is it necessary to set the schema here?

Reply:
in buildPhysicalJoinSchema, it will use `leftSchema`(because it is semi join) + `PhysicalJoin.schema.last()`(need match-helper).
```golang
newSchema := leftSchema.Clone()
newSchema.Append(join.Schema().Columns[join.Schema().Len()-1])
```

So we must add match-helper to physicalJoin before this function.